### PR TITLE
Fix/IAP fixes 

### PR DESCRIPTION
**Pull request by deeonwuli**
### :pushpin: References

-   **Issue:** Closes https://app.clickup.com/t/8696j3edq

### :memo: Implementation
- [x] Remove disease/hazard type from dropdown when event is completed
- [x] Persist existing disease/hazard types on form page reload
- [x] Add confirmation dialog on event completion click
- [x] Sort incident response action rows by due date
- [x] Blank out the loader background (with a grey background)
- [x] Get timeline based on due date
- [x] Include disease outbreak event id in page URL
- [x] Use incident response officers user group for response action Responsible officers dropdown field

### :fire: Notes to the tester

#8696j3edq

### CODE REVIEW

File: src/domain/usecases/GetConfigurationsUseCase.ts [hunk 1]
```diff
@@ -12,39 +12,47 @@ export class GetConfigurationsUseCase {
...
-                            return Future.success(configurations);
-                        });
-                });
-            });
-        });
+        return Future.joinObj({
+            allTeamMembers: this.teamMemberRepository.getAll(),
+            incidentResponseOfficers: this.teamMemberRepository.getIncidentResponseOfficers(),
+            managers: this.teamMemberRepository.getIncidentManagers(),
+            riskAssessors: this.teamMemberRepository.getRiskAssessors(),
```
Comment (9sneha-n) [inline src/domain/usecases/GetConfigurationsUseCase.ts:19]: nice

File: src/webapp/pages/form-page/useForm.ts [hunk 2]
```diff
@@ -67,9 +68,18 @@ export function useForm(formType: FormType, id?: Id): State {
...
     const currentEventTracker = getCurrentEventTracker();
     const { existingEventTrackerTypes } = useExistingEventTrackerTypes();
+    const { dataPerformanceOverview } = usePerformanceOverview();
     useCheckWritePermission(formType);
     const snackbar = useSnackbar();
 
+    const allDataPerformanceEvents = dataPerformanceOverview?.map(
+        event => event.hazardType || event.suspectedDisease
+    );
+    const existingEventTrackers =
```
Comment (9sneha-n) [inline src/webapp/pages/form-page/useForm.ts:78]: why is this required? useExistingEventTrackerTypes hook should be getiing the dataPerformanceOverview already. Is there some scenario this does not work ?

File: src/webapp/components/form/form-summary/EventTrackerFormSummary.tsx [hunk 3]
```diff
@@ -13,46 +13,33 @@ import { FormSummaryData } from "../../../pages/event-tracker/useDiseaseOutbreak
...
+    const { id, formType, formSummary, onOpenModal: onCompleteClick, globalMessage } = props;
     const { goTo } = useRoutes();
     const snackbar = useSnackbar();
 
     useEffect(() => {
-        if (!summaryError) return;
+        if (!globalMessage) return;
 
-        snackbar.error(summaryError);
-        goTo(RouteName.DASHBOARD);
```
Comment (9sneha-n) [inline src/webapp/components/form/form-summary/EventTrackerFormSummary.tsx:37]: the "  goTo(RouteName.DASHBOARD);" has been removed. I think it is required no?

**Comment by deeonwuli** [comment]
> looks good @deeonwuli One query on dataPerformanceOverview. Which scenario is this required? Is it on reload?  

yes, exactly that since the ` eventTrackerTypes` defaults back to an [] when the form page is reloaded

**Comment by 9sneha-n** [comment]
> > looks good @deeonwuli One query on dataPerformanceOverview. Which scenario is this required? Is it on reload?
> 
> yes, exactly that since the ` eventTrackerTypes` default back to an empty array when the form page is reloaded

ah nice one @deeonwuli

**Review by 9sneha-n — APPROVED** [review]
